javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/JavaParserTypeDeclarationAdapter.java

```diff
@@ -155,12 +156,15 @@ private ResolvedTypeDeclaration checkAncestorsForType(String name, ResolvedRefer
                 }
             }
         }
-        // We want to avoid infinite recursion when a class is using its own method
-        // see issue #75
+
         if (candidateMethods.isEmpty()) {
-            SymbolReference<ResolvedMethodDeclaration> parentSolution = context.getParent().solveMethod(name, argumentsTypes, staticOnly);
-            if (parentSolution.isSolved()) {
-                candidateMethods.add(parentSolution.getCorrespondingDeclaration());
+            // Search for static imports
+            Optional<CompilationUnitContext> cu = findCompilationUnitContext();
```

Comment:
Really minor and not something worth blocking this over, but would you mind renaming this to `cuCtx` or similar to indicate that it's not actually a compilation unit please?